mlir/lib/Util/Util.cpp

```diff
@@ -1248,6 +1248,12 @@ air::writeAccessPattern(memref::SubViewOp subview) {
     else
       std::get<1>(pattern).push_back(*subview_sizes++);
   }
+  assert(static_sizes.size() >= static_strides.size() &&
+         "only support standard and rank-reduced memref.subview");
```

Comment:
It looks like ultimately `writeAccessPattern` is called from `ShrinkMemrefSizesByAccessPattern` which could return failure + diagnostics instead of asserting (in Debug mode) or crashing in unexplained ways (in Release mode).

Reply:
Make sense. I'll PR later today to rewrite that with emitOpError.

Reply:
Fixed in https://github.com/Xilinx/mlir-air/pull/632